internal/outgoing/dispatcher.go

```diff
@@ -161,23 +189,45 @@ func (d *Dispatcher) processDelayedEvent(e models.IncomingEvent, timeRemain time
 		panic(storeErr)
 	}
 
-	jsonString, jsonErr := json.Marshal(e)
-	if jsonErr != nil {
-		panic(jsonErr)
-	}
-
+	var storedEvent models.IncomingEvent
 	if dataExists {
-		storedEvent := models.IncomingEvent{}
+		storedEvent = models.IncomingEvent{}
 		json.Unmarshal([]byte(storedData), &storedEvent)
-		if getControlTimestamp(storedEvent, 0) > getControlTimestamp(e, uint64(time.Now().UnixNano())) {
-			// Skip updating event data as stored data has newer timestamp
-			dLogger.WithFields(log.Fields{
-				"storedEvent":  storedEvent,
-				"event":        e.GetTraceInfo(),
-				"eventDataKey": "dataKey",
-			}).Debug("Skipping update on event data")
-			return
+	}
+
+	if dataExists && getControlTimestamp(storedEvent, 0) > getControlTimestamp(e, uint64(time.Now().UnixNano())) {
+		// Skip updating event data as stored data has newer timestamp
+		dLogger.WithFields(log.Fields{
+			"storedEvent":  storedEvent,
+			"event":        e.GetTraceInfo(),
+			"eventDataKey": "dataKey",
+		}).Debug("Skipping update on event data")
+		return
+	}
+
+	if dest.Config.KeepThrottledPayloads {
+		customizedPayload := d.customizePayload(&e, dest)
+		queueKey := getEventThrottledPayloadsKey(store, e, dest)
+		jsonString, jsonErr := json.Marshal(customizedPayload)
+		if jsonErr != nil {
+			panic(jsonErr)
 		}
+		store.Enqueue(queueKey, string(jsonString), dest.Config.GetThrottleValue()*2)
+	}
+
+	if dest.Config.IncludeDocument && dest.Config.KeepThrottledDocuments {
+		customizedDocument := d.customizeDocument(&e, dest, documentStore)
+		queueKey := getEventThrottledDocumentsKey(store, e, dest)
+		jsonString, jsonErr := json.Marshal(customizedDocument)
+		if jsonErr != nil {
+			panic(jsonErr)
+		}
+		store.Enqueue(queueKey, string(jsonString), dest.Config.GetThrottleValue()*2)
```

Comment:
does that mean if i enabled keep throttled docs & throttle payload, the event would be enqueued twice?

Reply:
oh nvm, looks like i'm confused by the word "enqueue" as they are actually 2 diff redis key just keeping the payloads and docs separately

Reply:
@tinyjacky `jsonString` here and above are different.
one is the `payload`, another is the `document`
I can change the variable name to be more specific.

Reply:
Yes I am now able to notice that. I think you can keep it as is